File: src/main/kotlin/limia/Exception/InvalidParametersException.java
package limia.Exception;

import java.util.ArrayList;

/**
 * Created by workstation on 28/04/2017.
 */
public class InvalidParametersException extends Exception {

    private ArrayList<String> parameters;

    public ArrayList<String> getParameters() {
        return parameters;
    }

    public void setParameters(ArrayList<String> parameters) {
        this.parameters = parameters;
    }

    public InvalidParametersException() {
    }

    public InvalidParametersException(String message) {
        super(message);
    }

    public InvalidParametersException(String message, Throwable cause) {
        super(message, cause);
    }

    public InvalidParametersException(Throwable cause) {
        super(cause);
    }

    public InvalidParametersException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
    }
}

File: src/main/kotlin/limia/Exception/InvalidParametersException.kt
package limia.Exception

import java.util.ArrayList

/**
 * Created by workstation on 28/04/2017.
 */
class InvalidParametersException : Exception {

    var parameters: ArrayList<String>? = null

    fun addParameter(parameter: String) {
        if (parameters == null){
            parameters = ArrayList<String>()
        } else {
            parameters!!.add(parameter)
        }

    }

    constructor()

    constructor(message: String) : super(message)

    constructor(message: String, cause: Throwable) : super(message, cause)

    constructor(cause: Throwable) : super(cause)

    constructor(message: String, cause: Throwable, enableSuppression: Boolean, writableStackTrace: Boolean) :
            super(message, cause, enableSuppression, writableStackTrace)
}
